### PR TITLE
[FLINK-23634][tests] Explicitly trigger scheduled tasks

### PR DESCRIPTION
**Pull request by rkhachatryan**
## What is the purpose of the change

Fix flaky `BatchingStateChangeUploaderTest.testDelay`
by explicitly checking and triggering scheduled tasks
instead of waiting.

### CODE REVIEW

File: flink-dstl/flink-dstl-dfs/src/test/java/org/apache/flink/changelog/fs/BatchingStateChangeUploaderTest.java [hunk 1]
```diff
@@ -94,14 +97,23 @@ public void testSizeThreshold() throws Exception {
...
                 delayMs,
                 Integer.MAX_VALUE,
+                scheduler,
                 (store, probe) -> {
+                    scheduler.triggerAll();
                     List<StateChangeSet> changeSets = getChanges(4);
                     upload(store, changeSets);
                     assertTrue(probe.getUploaded().isEmpty());
-                    Thread.sleep(delayMs * 2);
+                    assertTrue(
```
Comment (Myasuka) [inline flink-dstl/flink-dstl-dfs/src/test/java/org/apache/flink/changelog/fs/BatchingStateChangeUploaderTest.java:111]: Is this assert statemanet must exist here?

Comment (rkhachatryan) [reply flink-dstl/flink-dstl-dfs/src/test/java/org/apache/flink/changelog/fs/BatchingStateChangeUploaderTest.java:111]: The intention is to check that the delay, that it is equal to what was configured.

**Review by Myasuka — APPROVED** [review]
This PR LGTM, thanks for the fix.

**Comment by rkhachatryan** [comment]
Thanks for the review.
The build failed but the failed phases succeeded in the [private build](https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=21596&view=results) (`finegrained_resource_management`, `python`).
All the failures are unrelated.
Merging.

**Pull request merged**